<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="org.apache.kafka.common.network.SelectorTest" tests="30" skipped="0" failures="0" errors="0" timestamp="2020-01-15T02:08:50" hostname="localhost" time="0.771">
  <properties/>
  <testcase name="testGracefulClose" classname="org.apache.kafka.common.network.SelectorTest" time="0.026"/>
  <testcase name="testSendLargeRequest" classname="org.apache.kafka.common.network.SelectorTest" time="0.007"/>
  <testcase name="testLargeMessageSequence" classname="org.apache.kafka.common.network.SelectorTest" time="0.222"/>
  <testcase name="testLowestPriorityChannel" classname="org.apache.kafka.common.network.SelectorTest" time="0.005"/>
  <testcase name="testConnectDisconnectDuringInSinglePoll" classname="org.apache.kafka.common.network.SelectorTest" time="0.099"/>
  <testcase name="testConnectException" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="registerFailure" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testMute" classname="org.apache.kafka.common.network.SelectorTest" time="0.004"/>
  <testcase name="testExpireConnectionWithPendingReceives" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testCantSendWithInProgress" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testOutboundConnectionsCountInConnectionCreationMetric" classname="org.apache.kafka.common.network.SelectorTest" time="0.006"/>
  <testcase name="testImmediatelyConnectedCleaned" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testExistingConnectionId" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testCantSendWithoutConnecting" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testCloseOldestConnection" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testServerDisconnect" classname="org.apache.kafka.common.network.SelectorTest" time="0.005"/>
  <testcase name="testMetricsCleanupOnSelectorClose" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testPartialSendAndReceiveReflectedInMetrics" classname="org.apache.kafka.common.network.SelectorTest" time="0.216"/>
  <testcase name="testWriteCompletesSendWithNoBytesWritten" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testIdleExpiryWithoutReadyKeys" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testConnectionsByClientMetric" classname="org.apache.kafka.common.network.SelectorTest" time="0.011"/>
  <testcase name="testInboundConnectionsCountInConnectionCreationMetric" classname="org.apache.kafka.common.network.SelectorTest" time="0.004"/>
  <testcase name="testNoRouteToHost" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testPartialReceiveGracefulClose" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testNormalOperation" classname="org.apache.kafka.common.network.SelectorTest" time="0.12"/>
  <testcase name="testMuteOnOOM" classname="org.apache.kafka.common.network.SelectorTest" time="0.003"/>
  <testcase name="testCloseOldestConnectionWithMultiplePendingReceives" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testExpireClosedConnectionWithPendingReceives" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testConnectionRefused" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <testcase name="testEmptyRequest" classname="org.apache.kafka.common.network.SelectorTest" time="0.002"/>
  <system-out><![CDATA[[2020-01-15 10:08:50,988] ERROR Reclaimed buffer of size 95446630 and identity 1262692894 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:50,988] ERROR Reclaimed buffer of size 10 and identity 1786066491 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:50,988] ERROR Reclaimed buffer of size 95446630 and identity 553385651 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:50,989] ERROR Reclaimed buffer of size 10 and identity 693951831 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:50,989] ERROR Reclaimed buffer of size 95446630 and identity 1139126865 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:50,989] ERROR Reclaimed buffer of size 10 and identity 352600992 that was not properly release()ed. This is a bug. (org.apache.kafka.common.memory.GarbageCollectedMemoryPool:106)
[2020-01-15 10:08:51,215] ERROR Unexpected exception during send, closing connection 0 and rethrowing exception {} (org.apache.kafka.common.network.Selector:406)
java.lang.IllegalStateException: Attempt to begin a send operation with prior send operation still in progress, connection id is 0
	at org.apache.kafka.common.network.KafkaChannel.setSend(KafkaChannel.java:377)
	at org.apache.kafka.common.network.Selector.send(Selector.java:398)
	at org.apache.kafka.common.network.SelectorTest.testCantSendWithInProgress(SelectorTest.java:164)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
